### PR TITLE
test commit

### PR DESCRIPTION
**Pull request by Vg4881035**
First pull request

### CODE REVIEW

File: Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+	
+	@Override
+	public User[] retrieve() throws Exception {
+
+		long lineCount = Files.lines(Paths.get("hometask.txt")).filter(line -> !line.startsWith("#")).count();
+		User[] converted = new User[(int) lineCount];
+
+		Scanner sc = new Scanner("hometask.txt");
+		int i = 0;
```
Comment (ViacheslavButov) [inline Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java:23]: why do you need this variable?

File: Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		}
+		sc.close();
+
+		return converted;
+
+	}
+
+	@Override
+	public UserConverter getConverter() {
+		return userConverter = new UserConverterImpl();
```
Comment (ViacheslavButov) [inline Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java:38]: no, you should return your class variable, not the new instance.

return userConverter;

**Comment by ViacheslavButov** [comment]
Please create .gitignore file and all the files and folders like *eclipse, *idea, *sonar move there

File: Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java (hunk 1 above)
Comment (Vg4881035) [reply Lesson4/src/source/it/butov/homework/UserRetrieverImpl.java:23]: int i = 0;

		while (sc.hasNextLine()) {
			String line = sc.nextLine();
			converted[i] = userConverter.convertFromCsv(line);
			i++;

Я использовал "i" для указания индекса в цикле.

**Pull request merged**